hydra-node/src/Hydra/API/ServerOutput.hs

```diff
@@ -2,15 +2,41 @@
 
 module Hydra.API.ServerOutput where
 
+import Data.Aeson (Value (..), withObject, (.:))
+import qualified Data.Aeson.KeyMap as KeyMap
 import Hydra.API.ClientInput (ClientInput (..))
 import Hydra.Chain (ChainStateType, IsChainState, PostChainTx, PostTxError)
 import Hydra.Crypto (MultiSignature)
 import Hydra.Ledger (IsTx, UTxOType, ValidationError)
 import Hydra.Network (NodeId)
 import Hydra.Party (Party)
-import Hydra.Prelude
+import Hydra.Prelude hiding (seq)
 import Hydra.Snapshot (Snapshot, SnapshotNumber)
 
+-- | The type of messages sent to clients by the 'Hydra.API.Server'.
+data TimedServerOutput tx = TimedServerOutput
+  { output :: ServerOutput tx
+  , seq :: Natural
+  , time :: UTCTime
+  }
+  deriving stock (Eq, Show, Generic)
+
+instance Arbitrary (ServerOutput tx) => Arbitrary (TimedServerOutput tx) where
+  arbitrary = genericArbitrary
+
+instance (ToJSON tx, IsChainState tx) => ToJSON (TimedServerOutput tx) where
+  toJSON TimedServerOutput{output, seq, time} =
+    case toJSON output of
+      Object o ->
+        Object $ o <> KeyMap.fromList [("seq", toJSON seq), ("timestamp", toJSON time)]
```

Comment:
maybe we should add a note saying:

warning: if `o` contains attributes defined for keys "seq" or "timestamp", they will get override by `KeyMap.fromList`

wdty? do you find it redundant to say?

Reply:
If you reach this point, the code should explain that already (all `<>` are right biased)